### PR TITLE
Modify signature of dequantize ops for decomposed quantized Tensor

### PR DESCRIPTION
**Pull request by kausikmaiti**
Note: The initial purpose of this PR is to draw suggestion and feedback regarding better alternative, if any.

At present, dequantize op for decomposed quantized Tensor representation e.g. dequantize_per_tensor() assumes the output dtype as torch.float and hence, it does not have the output dtype in its operator argument list. However, this op signature becomes unusable when the assumption breaks. Because, in case the output dtype is different from torch.float, there is no way to specify the same during dequantization.

This change is aimed at generalizing the signature of dequantize op like dequantize_per_tensor() for wider use-cases where the output dtype can be different from torch.float and needs to passed during dequantization. The proposal is to use an additional argument named 'output_dtype' to solve the problem. However, we would also like to have suggestion and feedback regarding any better alternative that can be used instead.

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

### CODE REVIEW

**Comment by kausikmaiti** [comment]
Hi, I'm looking forward to the review comments on this change. Kindly help in this regard. Thank you.

**Comment by jerryzh168** [comment]
yeah this makes sense, but it will break executorch right now I think, we need someone from our side to import this PR and make corresponding changes in executorch as well

File: torch/ao/quantization/fx/_decomposed.py [hunk 1]
```diff
@@ -208,7 +214,7 @@ def dequantize_per_tensor_tensor_meta(input, scale, zero_point, quant_min, quant
 # TODO: remove other variants and keep this one
 quantized_decomposed_lib.define(
     "dequantize_per_tensor.tensor2(Tensor input, Tensor scale, Tensor zero_point, "
-    "Tensor quant_min, Tensor quant_max, ScalarType dtype) -> Tensor")
+    "Tensor quant_min, Tensor quant_max, ScalarType dtype, ScalarType output_dtype) -> Tensor")
 
 @impl(quantized_decomposed_lib, "dequantize_per_tensor.tensor2", "CompositeExplicitAutograd")
 def dequantize_per_tensor_tensor2(
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_decomposed.py:220]: we might be able to remove this op, maybe it's better to remove first before adding this PR

**Comment by jerryzh168** [comment]
also @kausikmaiti please fix the tests

**Comment by jerryzh168** [comment]
@kausikmaiti just a heads up, this PR needs to be landed through phabricator since the final change has to touch both executorch and pytorch, so we'd like be closing this PR and opening another one after everything is ready to land.

**Comment by manuelcandales** [comment]
@kausikmaiti Could you make output_dtype a keyword argument with a default value?
I am working on the signature changes regardless. But by making it keyword argument with default value, all of the python code that depends on it, won't break.

**Comment by kausikmaiti** [comment]
> @kausikmaiti Could you make output_dtype a keyword argument with a default value? I am working on the signature changes regardless. But by making it keyword argument with default value, all of the python code that depends on it, won't break.

Hi, Thank you for the suggestions. I've made the changes as suggested. I will add the change as a separate commit after testing.

**Comment by manuelcandales** [comment]
Landed through PR: https://github.com/pytorch/pytorch/pull/121450

**Pull request closed without merging**